### PR TITLE
ogrdxflayer.cpp: use std::unique_ptr

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp [hunk 1]
```diff
@@ -459,7 +459,7 @@ OGRDXFFeature *OGRDXFLayer::TranslateMTEXT()
 {
     char szLineBuf[512];
     int nCode = 0;
-    OGRDXFFeature *poFeature = new OGRDXFFeature( poFeatureDefn );
+    auto poFeature = std::unique_ptr<OGRDXFFeature>(new OGRDXFFeature( poFeatureDefn ));
```
[the remark this answers is not in the document]
Comment (rouault) [reply gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp:462]: > Is there a reason not to use `std::make_unique`?

yes, this is C++14, and our base requirement is C++11 currently. The (optional) PDF and MongoDB driver have higher requirements due to the underlying libraries requiring C++14 themselves.

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp [hunk 2]
```diff
@@ -2190,21 +2143,20 @@ OGRDXFFeature *OGRDXFLayer::TranslateSPLINE()
 /*      Use the helper function to check the input data and insert      */
 /*      the spline.                                                     */
 /* -------------------------------------------------------------------- */
-    OGRLineString *poLS = InsertSplineWithChecks( nDegree,
-        adfControlPoints, nControlPoints, adfKnots, nKnots, adfWeights );
+    auto poLS = std::unique_ptr<OGRLineString>(InsertSplineWithChecks( nDegree,
+        adfControlPoints, nControlPoints, adfKnots, nKnots, adfWeights ));
```
Comment (atlight) [inline gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp:2147]: This function is defined in this file so it would make as much sense to change it to return `unique_ptr`. But maybe you want to do that in another commit.

**Review by atlight — APPROVED** [review]

File: gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp (hunk 2 above)
Comment (rouault) [reply gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp:2147]: yeah, that can be done later. I just wanted to fix a few things on which cppcheck stumbled upon (false positives), and did similar changes through that file.

**Comment by rouault** [comment]
follow-up in https://github.com/OSGeo/gdal/pull/4226 adding a cpl::make_unique<> C++11 compatible equivalent of std::make_unique
